src/main/java/androidx/text/SpannableStringBuilder.kt

```diff
@@ -134,3 +138,39 @@ inline fun SpannableStringBuilder.scale(
     proportion: Float,
     builderAction: SpannableStringBuilder.() -> Unit
 ) = inSpans(RelativeSizeSpan(proportion), builderAction = builderAction)
+
+/**
+ * Wrap appended text in [builderAction] in a [SuperscriptSpan].
+ *
+ * @see SpannableStringBuilder.inSpans
+ */
+inline fun SpannableStringBuilder.up(builderAction: SpannableStringBuilder.() -> Unit) =
+    inSpans(SuperscriptSpan(), builderAction = builderAction)
+
+/**
+ * Wrap appended text in [builderAction] in a [SubscriptSpan].
+ *
+ * @see SpannableStringBuilder.inSpans
+ */
+inline fun SpannableStringBuilder.down(builderAction: SpannableStringBuilder.() -> Unit) =
+    inSpans(SubscriptSpan(), builderAction = builderAction)
+
+/**
+ * Wrap appended text in [builderAction] in a [TypefaceSpan].
+ *
+ * @see SpannableStringBuilder.inSpans
+ */
+inline fun SpannableStringBuilder.font(
```

Comment:
I'm torn between calling this "font" or "typeface" to match the span name. I suspect matching the span name will improve discoverability.

Reply:
Matching name seems like better, I renamed the methods - "up", "down", "font" to "superscript", "subscript" and "typeface" .